client/state/route/types.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import { ROUTE_CLEAR_LAST_NON_EDITOR, ROUTE_SET } from 'calypso/state/action-types';
+
+export type Query = {
+	[ key: string ]: string | number | undefined;
+	_timestamp?: number;
+};
```

Comment:
Not fun of this type. Hopefully, there's a way to express it differently. `_timestamp` is the only property of type `number` that may also be undefined - everything else is a `string`.